src/map/lua/luautils.cpp

```diff
@@ -2331,7 +2331,10 @@ int32 OnMobDeath(CBaseEntity* PMob, CBaseEntity* PKiller)
 
     CLuaBaseEntity LuaMobEntity(PMob);
 
-    luautils::OnMobDeathEx(PMob, PKiller, true);
+    if (PChar)

```

Comment:
this doesn't work the way you think it does - casting it won't let you know if the object type is actually that type, it only forces the program to interpret that memory location as if it were a character - it'll still crash unless you do PKiller->objtype == TYPE_PC